transformer_engine/pytorch/module/base.py

```diff
@@ -684,16 +694,22 @@ def grad_output_preprocess(
                 grad_output_c = ctx.ub_obj_gradout.get_ubuf_output(0)
             else:
                 grad_output_c = torch.empty_like(grad_output_mat, dtype=torch.uint8)
-            cast_to_fp8(
-                grad_output_mat,
-                ctx.fp8_meta["scaling_bwd"],
-                tex.FP8BwdTensors.GRAD_OUTPUT1,
-                fp8_dtype_backward,
-                out=grad_output_c,
-            )
+            if not isinstance(grad_output_mat, Float8Tensor):
+                cast_to_fp8(
+                    grad_output_mat,
+                    ctx.fp8_meta["scaling_bwd"],
+                    tex.FP8BwdTensors.GRAD_OUTPUT1,
+                    fp8_dtype_backward,
+                    out=grad_output_c,
+                )
+            else:
+                grad_output_c = grad_ouput_mat # pylint: disable=undefined-variable
```

Comment:
@cyanguwa This is a typo. It should be `grad_output_mat`.

Reply:
good catch - we will fix it.

Reply:
https://github.com/NVIDIA/TransformerEngine/pull/795